tunnel/tunnel.go

```diff
@@ -91,22 +92,33 @@ func Find(ipAddrs ...netip.Addr) (Tunnel, error) {
 	return Tunnel{}, ErrNotFound
 }
 
+func addDelAddr(cmd string, ifaceName string, addr string) ([]byte, error) {
+	// #nosec G204 -- input is properly sanitized
+	cmdHandle := exec.Command(
+		"ip",
+		"address",
+		cmd,
+		addr,
+		"dev",
+		ifaceName,
+	)
+	return cmdHandle.CombinedOutput()
+}
+
 func (t *Tunnel) cmdAddrs(cmd string) error {
 	for _, ip := range t.ips {
 		mask := 10 // unify with other platforms
 		if ip.BitLen() > 32 {
 			mask = ip.BitLen() // ipv6
 		}
-		// #nosec G204 -- input is properly sanitized
-		cmd := exec.Command(
-			"ip",
-			"address",
-			cmd,
-			fmt.Sprintf("%s/%d", ip.String(), mask),
-			"dev",
-			t.iface.Name,
-		)
-		out, err := cmd.CombinedOutput()
+		out, err := addDelAddr(cmd, t.iface.Name, fmt.Sprintf("%s/%d", ip.String(), mask))
+		if err != nil {
+			return fmt.Errorf("%s IP address to interface: %s : %w", cmd, string(out), err)
+		}
+	}
+
+	for _, prefix := range t.prefixes {
```

Comment:
`ips` and `prefixes` are mutually exclusive? there is no protection against that